dev-utils/webdriver.js

```diff
@@ -158,6 +167,101 @@ function getWebdriveBaseConfig(path) {
   }
 }
 
+function waitForApmServerCalls(errorCount = 0, transactionCount = 0) {
+  console.log(
+    `Waiting for minimum ${errorCount} Errors and ${transactionCount} Transactions.`
+  )
+  const serverCalls = browser.executeAsync(
+    function(errorCount, transactionCount, done) {
+      var apmServerMock = window.elasticApm.serviceFactory.getService(
+        'ApmServer'
+      )
+
+      function checkCalls() {
+        var serverCalls = apmServerMock.calls
+
+        var validCalls = true
+
+        if (errorCount) {
+          validCalls =
+            validCalls &&
+            serverCalls.sendErrors &&
+            serverCalls.sendErrors.length >= errorCount
+        }
+        if (transactionCount) {
+          validCalls =
+            validCalls &&
+            serverCalls.sendTransactions &&
+            serverCalls.sendTransactions.length >= transactionCount
+        }
+
+        if (validCalls) {
+          console.log('calls', serverCalls)
+          var promises = []
+
+          if (serverCalls.sendErrors) {
```

Comment:
this is not required, we already check for the same value in validCalls which covers it.

Reply:
Okay, now i get why we need it since we use the same var for both error and trans.

Let's refactor this code a bit.. checks for errors and transactions is repetitive in lots of places.

Reply:
Let's refactor this separately!